### PR TITLE
Add conn.interrupt() to DuckDB python API

### PR DESCRIPTION
**Pull request by henrinikku**
Implements https://github.com/duckdb/duckdb/issues/5938.

### CODE REVIEW

File: tools/pythonpkg/tests/fast/api/test_connection_interrupt.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import duckdb
+import pytest
+
+class TestConnectionInterrupt(object):
+    def test_connection_interrupt(self):
+        conn = duckdb.connect()
+        conn.execute("create table t (i integer)")
+        conn.execute("insert into t values (1)")
+        conn.interrupt()
```
Comment (Mause) [inline tools/pythonpkg/tests/fast/api/test_connection_interrupt.py:9]: Not sure this test is particularly useful? This is the one I implemented for the java version: https://github.com/duckdb/duckdb/pull/7489/files#diff-723bbb48d2778196c6ff18921818c3f0be362c3fd461f690682eec8c9f37dee0R3565

Comment (Tishj) [reply tools/pythonpkg/tests/fast/api/test_connection_interrupt.py:9]: Maybe using a custom UDF to make the execution sleep and communicating to another thread which then calls `interrupt` would be an easy way to test

Comment (Tishj) [reply tools/pythonpkg/tests/fast/api/test_connection_interrupt.py:9]: We also have a test for keyboard interrupt
`tools/pythonpkg/tests/fast/api/test_query_interrupt.py`

Maybe you can just make a variation on that, that might be easiest

Comment (henrinikku) [reply tools/pythonpkg/tests/fast/api/test_connection_interrupt.py]: Done. Thanks for providing the examples, I was looking for something similar but didn't notice `test_query_interrupt.py`.

**Review by Tishj — APPROVED** [review]
Looks good to me!

The description of `interrupt` could be a little better but overall I think it's fine

**Pull request merged**